compact.go

```diff
@@ -668,23 +670,20 @@ func (c *LeveledCompactor) populateBlock(blocks []BlockReader, mint int64, maxt
 		}
 
 		for i, chk := range chks {
-			// The way a populateBlock is used does not assume chunks before the requested min time.
-			// We can add support for that once it is needed.
+			// Current usecases don't require handling chunks before the requested min time.
 			if chk.MinTime < mint {
 				return BlockStats{}, errors.Errorf("found chunk with minTime: %d maxTime: %d outside of compacted minTime: %d maxTime: %d",
 					chk.MinTime, chk.MaxTime, mint, maxt)
 			}
 
-			// It can happen that we get here chunk partially or completely outside of requested meta.MaxTime e.g
-			// when we snapshot from head block which is also used for appends concurrently to this method.
-			if chk.MinTime > maxt {
-				// Ignore block that is completely outside.
+			// During snapshotting an append to a chunk can happen after requesting mint,maxt
+			// so data outside the expected time range should to be removed
```

Comment:
still not sure if this comment is completely clear so if you think of something else can update it.